src/formatters/deployResultFormatter.ts

```diff
@@ -270,6 +270,21 @@ export class DeployResultFormatter extends TestResultsFormatter implements Forma
     if (this.result.response.status === RequestStatus.Succeeded) return;
 
     const failures = this.relativeFiles.filter(isSdrFailure);
+    const deployMessages = ensureArray(this.result.response.details?.componentFailures);
+    if (deployMessages.length > failures.length) {
+      // if there's additional failures in the API response, find the failure and add it to the output
+      deployMessages.map((deployMessage) => {
+        if (!failures.some((f) => f.type === deployMessage.componentType && f.fullName === deployMessage.fullName)) {
```

Comment:
there's sometimes thousands of failures (chain reaction) and this'll be bigO-squared.  So maybe we build a set of `f.type#f.fullName` and use set.has 
```suggestion
      const failuresSet = new Set<string>(failures.map(f => `${f.type}#${f.fullName}`
      deployMessages.map((deployMessage) => {
        if (!failuresSet.has(`${deployMessage.componentType}#${deployMessage.fullName}`) {
```

there's also a fn for the key already if you want to move/expose it and only have 1 instead of 3 🤷🏻 https://github.com/salesforcecli/plugin-deploy-retrieve/blob/a916d3c5fba458b2d0c1b2397f59ceda6f9b5def/src/utils/previewOutput.ts#L100-L101